beetsplug/edit.py

```diff
@@ -358,6 +374,12 @@ def before_choose_candidate_listener(self, session, task):
 
         return choices
 
+    def import_begin_listener(self, session):
+        """Load data from import session into self"""
+        self.readonly_fields = {}
+        for field, view in config["import"]["set_fields"].items():
+            self.readonly_fields[field] = view.get()
```

Comment:
Is there a specific reason that you used a separate event handler here? It also seems like we could fetch the list of fields directly in the `edit_object` method.

Reply:
no specific reason, just bad planning (I initially thought that the set fields were only accessible in the `session` object, which is out of scope in the `edit_object` method, and then didn't simplify the code when I found out it wasn't the case)